plugins/FirmwareUpdater/FirmwareUpdaterMachineAction.qml

```diff
@@ -0,0 +1,191 @@
+// Copyright (c) 2018 Ultimaker B.V.
+// Cura is released under the terms of the LGPLv3 or higher.
+
+import QtQuick 2.2
+import QtQuick.Controls 1.1
+import QtQuick.Layouts 1.1
+import QtQuick.Window 2.1
+import QtQuick.Dialogs 1.2 // For filedialog
+
+import UM 1.2 as UM
+import Cura 1.0 as Cura
+
+
+Cura.MachineAction
+{
+    anchors.fill: parent;
+    property bool printerConnected: Cura.MachineManager.printerConnected
+    property var activeOutputDevice: printerConnected ? Cura.MachineManager.printerOutputDevices[0] : null
+    property var canUpdateFirmware: activeOutputDevice ? activeOutputDevice.activePrinter.canUpdateFirmware : false
+
+    Column
+    {
+        id: firmwareUpdaterMachineAction
+        anchors.fill: parent;
+        UM.I18nCatalog { id: catalog; name:"cura"}
+        spacing: UM.Theme.getSize("default_margin").height
+
+        Label
+        {
+            width: parent.width
+            text: catalog.i18nc("@title", "Update Firmware")
+            wrapMode: Text.WordWrap
+            font.pointSize: 18
+        }
+        Label
+        {
+            width: parent.width
+            wrapMode: Text.WordWrap
+            text: catalog.i18nc("@label", "Firmware is the piece of software running directly on your 3D printer. This firmware controls the step motors, regulates the temperature and ultimately makes your printer work.")
```

Comment:
I'm not a 100% sure if it can happen, but if we set a width, we should also set elide, right?

Reply:
No, because wrapMode is set to Text.WordWrap, so this Label renders multiple lines and will not elide.